src/network/net_addons.cc

```diff
@@ -0,0 +1,313 @@
+
+		result_vector.push_back(info);
+	}
+
+	return result_vector;
+}
+
+static void check_downloaded_file(const std::string& path, const std::string& checksum) {
+	try {
+		// Our md5 implementation is not well documented, so I am doing this
+		// as it is done in GameClient::handle_new_file and hope it works…
+		FileRead fr;
+		fr.open(*g_fs, path);
+		const size_t bytes = fr.get_size();
+		std::unique_ptr<char[]> complete(new char[bytes]);
+		fr.data_complete(complete.get(), bytes);
+		SimpleMD5Checksum md5sum;
+		md5sum.data(complete.get(), bytes);
+		md5sum.finish_checksum();
+		const std::string md5 = md5sum.get_checksum().str();
+		if (checksum != md5) {
+			throw wexception("Downloaded file '%s': Checksum mismatch, found %s, expected %s",
+			                 path.c_str(), md5.c_str(), checksum.c_str());
+		}
+	} catch (const std::exception& e) {
+		throw wexception(
+		   "Downloaded file '%s': Unable to check output file: %s", path.c_str(), e.what());
+	}
+}
+
+// TODO(Nordfriese): Add-on downloading speed would benefit greatly from storing
+// the files as ZIPs on the server. Similar for translation bundles. Perhaps
+// someone would like to write code to uncompress a downloaded ZIP file some day…
+
+void NetAddons::download_addon_file(const std::string& name,
+                                    const std::string& checksum,
+                                    const std::string& output) {
+	init();
+
+	set_url_and_timeout(
+	   std::string("https://raw.githubusercontent.com/widelands/wl_addons_server/master/addons/") +
+	   name);
+
+	std::FILE* out_file = std::fopen(output.c_str(), "wb");
+	curl_easy_setopt(curl_, CURLOPT_WRITEFUNCTION,
+	                 [](void* ptr, size_t size, size_t nmemb, std::FILE* stream) {  // NOLINT
+		                 return std::fwrite(ptr, size, nmemb, stream);
+	                 });
+	curl_easy_setopt(curl_, CURLOPT_WRITEDATA, out_file);
+
+	const CURLcode res = curl_easy_perform(curl_);
+
+	fclose(out_file);
+
+	if (res != CURLE_OK) {
+		throw wexception("%s: CURL terminated with error code %d", name.c_str(), res);
+	}
+	check_downloaded_file(output, checksum);
+}
+
+std::string NetAddons::download_i18n(const std::string& name,
+                                     const std::string& checksum,
+                                     const std::string& locale) {
+	init();
+
+	const std::string temp_dirname =
+	   kTempFileDir + FileSystem::file_separator() + name + ".mo" + kTempFileExtension;
+	g_fs->ensure_directory_exists(temp_dirname);
+
+	const std::string relative_output =
+	   temp_dirname + FileSystem::file_separator() + locale + kTempFileExtension;
+	const std::string canonical_output =
+	   g_fs->canonicalize_name(i18n::get_homedir() + "/" + relative_output);
+
+	set_url_and_timeout(
+	   std::string("https://raw.githubusercontent.com/widelands/wl_addons_server/master/i18n/") +
+	   name + "/" + locale);
+
+	std::FILE* out_file = std::fopen(canonical_output.c_str(), "wb");
```

Comment:
Try using FileWrite here, so you don't have to handle absolute file paths here. This causes troubles on Windows.